R/github.R

```diff
@@ -305,17 +305,54 @@ check_no_github_repo <- function(owner, repo, host, auth_token) {
   ui_stop("Repo {ui_value(spec)} already exists on {ui_value(where)}.")
 }
 
-check_github_token <- function(auth_token) {
-  if (is.null(auth_token) || !nzchar(auth_token)) {
+
+# github token helpers ----------------------------------------------------
+
+## minimal check: token is not ""
+have_github_token <- function(auth_token = github_token()) {
+  !isTRUE(auth_token == "")
+}
+
+## tolerate "", but explicitly validate the token otherwise
+maybe_github_token <- function(auth_token = github_token()) {
+  if (!have_github_token(auth_token)) {
+    return("")
+  }
+  validate_github_token(auth_token)
+}
+
+## rigorous checks, including against the GitHub API
+validate_github_token <- function(auth_token = github_token()) {
```

Comment:
I think this should be `check_github_token()` for consistency with the other `check_` functions (which throw an error)

Reply:
Yeah, this was a leftover from experimenting with two different levels of checks.